covasim/sim.py

```diff
@@ -370,17 +373,18 @@ def next(self, verbose=0):
                         community_contact_inds = cvu.choose(max_n=pop_size, n=n_comm_contacts)
                         person_contacts['c'] = community_contact_inds
 
+                    
                     # Determine who gets infected
                     for ckey in self.contact_keys:
                         contact_ids = person_contacts[ckey]
                         if len(contact_ids):
-                            this_beta_layer = thisbeta*beta_layers[ckey]
-                            transmission_inds = cvu.bf(this_beta_layer, contact_ids)
+                            contacts = self.people.get_list(contact_ids)
+                            this_beta_layer = thisbeta * beta_layers[ckey]
+                            transmission_inds = cvu.bfl(rel_susceptibility[contact_ids] * this_beta_layer, contact_ids)
                             for contact_ind in transmission_inds: # Loop over people who get infected
                                 target_person = self.people[contact_ind]
-                                if target_person.susceptible: # Skip people who are not susceptible
-                                    new_infections += target_person.infect(t, bed_constraint, source=person) # Actually infect them
-                                    sc.printv(f'        Person {person.uid} infected person {target_person.uid}!', 2, verbose)
+                                new_infections += target_person.infect(t, bed_constraint, source=person) # Actually infect them
```

Comment:
Why was `if target_person.susceptible:` removed?

Reply:
rel_susceptibility has 0 for non susceptible, so this check is met by line 383